network/peermgr.go

```diff
@@ -157,3 +181,39 @@ func (mgr *peerMgr) CheckConnectivity() {
 		}
 	}
 }
+
+type PeerStore struct {
```

Comment:
Why we can't use peerMgr?

Reply:
Peer ID's will be saved in Unicode format and make problems.

Reply:
This `lp2ppeer.ID` type makes a problem for us. Peer IDs going to be saved as the unicodes and then we can't retrive them at all. we can use `map[string]PeerInfo` without defineing it as a type ot struct. but the main problem is json tags.